### ISSUE TITLE
BUG: Nanquantile doens't handle multidimensional `q` correctly

### ISSUE THREAD

**Issue by seberg** [numpy/numpy#25731]
Quantile/percentile insert the dimensions of `q` at the beginning of the result.  Nanquantile seems not to be written to correctly support multi-dimensional `q` and ends up splitting it up:
```python
a = np.ones((3, 4))
q = np.full((1, 2), 0.5)
 
print(np.quantile(a, q, axis=1).shape)
# (1, 2, 3)  correct
print(np.nanquantile(a, q, axis=1).shape)
# (1, 3, 2) mangled up
```
The axes contributed by `q` of shape `(1, 2)` should end up at the beginning but get split.

(Reported by @aschaffer)

**Comment by aschaffer** [comment]
`nanquantile()` seems to insert `q` dimensions into the selected `axis`, instead of at the beginning, like `quantile/percentile` do.

**Comment by seberg** [comment]
Fixed in 2.0 now.

**Comment by aschaffer** [comment]
[like]  Andrei Schaffer reacted to your message:
________________________________
